### PR TITLE
test(crux-ui): image config, projects, teams, storage tests

### PR DESCRIPTION
**Pull request by ngyradi**
Added tests for: Dashboard, Image config, Projects, Storage, Teams.

### CODE REVIEW

File: web/crux-ui/e2e/utils/websocket-match.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
+export const wsPatchMatchPorts = (internalPort: string, externalPort?: string) => (payload: any) => {
+  const internal = Number.parseInt(internalPort, 10)
+  const external = Number.parseInt(externalPort, 10)
+
+  return payload.config?.ports?.some(it => it.internal === internal && (!external || it.external === external))
+}
+
+export const wsPatchMatchPortRange =
+  (internalFromPort: string, externalFromPort?: string, internalToPort?: string, externalToPort?: string) =>
```
Comment (robot9706) [inline web/crux-ui/e2e/utils/websocket-match.ts:9]: Only the `internalFromPort` is required?

File: web/crux-ui/e2e/utils/websocket-match.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+export const wsPatchMatchExpose = (expose: string) => (payload: any) => {
+  return payload.config?.expose === expose
+}
+
+export const wsPatchMatchTTY = (tty: boolean) => (payload: any) => {
+  return payload.config?.tty === tty
+}
+
+export const wsPatchMatchEnvironment = (key: string, value: string) => (payload: any) => {
+  return payload.config?.environment[0]?.key === key && payload.config?.environment[0]?.value === value
```
Comment (robot9706) [inline web/crux-ui/e2e/utils/websocket-match.ts:65]: This should check all environment variables, not just the first.

File: web/crux-ui/e2e/utils/websocket-match.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    cmd: string,
+    envKey: string,
+    envVal: string,
+  ) =>
+  (payload: any) => {
+    let init = payload.config?.initContainers[0]
+    return (
+      init?.name === name &&
+      init?.image === image &&
+      init?.volumes[0].name === volName &&
```
Comment (robot9706) [inline web/crux-ui/e2e/utils/websocket-match.ts:90]: Same as environment.

File: web/crux-ui/e2e/utils/websocket-match.ts [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  let storage = payload.config?.storage
+  return storage?.storageId === storageId && storage?.bucket === bucketPath && storage?.path === volume
+}
+
+//docker
+export const wsPatchMatchLogConfig = (driver: string, key?: string, value?: string) => (payload: any) => {
+  let logCfg = payload.config?.logConfig
+  return (
+    logCfg?.driver === driver &&
+    (!key || logCfg?.options[0]?.key === key) &&
```
Comment (robot9706) [inline web/crux-ui/e2e/utils/websocket-match.ts:115]: Same as environment.

File: web/crux-ui/e2e/with-login/dashboard.spec.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { expect, test } from '@playwright/test'
+
+test.describe('Onboarding', () => {
+  test('should be toggleable', async ({ page }) => {
+    await page.goto(ROUTE_DASHBOARD)
+    await page.waitForURL(ROUTE_DASHBOARD)
+    await expect(page.locator('label:has-text("Onboarding")')).toBeVisible()
+
+    await page.goto(ROUTE_PROFILE)
+    await page.locator('button[role="switch"]').click()
```
Comment (robot9706) [inline web/crux-ui/e2e/with-login/dashboard.spec.ts:11]: I think this should also look for the label. If there are multiple toggles on the profile page this won't work anymore.

File: web/crux-ui/e2e/with-login/version.spec.ts [hunk 6]
```diff
@@ -31,9 +31,12 @@ test("New version should get the default version's images and deployments", asyn
 
   await page.goto(projectUrl(projectId))
 
+  //changed because the two projects would sometimes switch place when running e2e tests and fail
   await expect(page.locator('.bg-dyo-blue:has-text("Incremental")')).toHaveCount(2)
-  await expect(page.locator(`.bg-error-red:has-text("DEFAULT"):right-of(h5:has-text("1.0.0"))`)).toHaveCount(1)
-  await expect(page.locator(`.bg-error-red:has-text("DEFAULT"):right-of(h5:has-text("2.0.0"))`)).toHaveCount(0)
+  //await expect(page.locator(`.bg-error-red:has-text("DEFAULT"):right-of(h5:has-text("1.0.0"))`)).toHaveCount(1)
```
Comment (robot9706) [inline web/crux-ui/e2e/with-login/version.spec.ts:36]: Comments?

File: web/crux-ui/e2e/with-login/storage.spec.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  await page.locator('button:has-text("Save")').click()
+  await expect(page.locator(`div.card h3:has-text('${storageName}-edited')`)).toBeVisible()
+})
+
+test('Can edit storage url', async ({ page }) => {
+  const storageName = 'storage-url-edit-test'
+  const storageId = await createStorage(page, storageName, TEST_URL, TEST_ACCESS_KEY, TEST_SECRET_KEY)
+  await page.goto(storageUrl(storageId))
+  await page.locator('button:has-text("Edit")').click()
+  let newUrl = (await removeUrlDomainSuffix(TEST_URL)).concat(`.edited.${TEST_URL.split('.').pop()}`)
```
Comment (robot9706) [inline web/crux-ui/e2e/with-login/storage.spec.ts:58]: There's no need for `removeUrlDomainSuffix` to be async.

File: web/crux-ui/e2e/with-login/project.spec.ts [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  const projName = 'filter-by-name-test-project'
+  await createProject(page, projName, 'versionless')
+  await page.goto(ROUTE_PROJECTS)
+  await expect(page.locator('h3:has-text("Filters")')).toBeVisible()
+  await expect(page.locator('button.bg-dyo-turquoise:has-text("All")')).toBeVisible()
+  await page.locator('input[placeholder="Search"] >> visible=true').fill(projName)
+  await expect(page.getByAltText('Picture of ', { exact: false })).toHaveCount(1)
+})
+
+//test.describe.configure({ mode: 'parallel' })
```
Comment (robot9706) [inline web/crux-ui/e2e/with-login/project.spec.ts:72]: Remove comment please.

File: web/crux-ui/e2e/with-login/nodes-deploy.spec.ts [hunk 9]
```diff
@@ -12,7 +12,7 @@ import {
 
 const PREFIX = 'first'
 const PREFIX_TWO = 'second'
-const IMAGE = 'nginx'
+const IMAGE = NGINX_TEST_IMAGE_WITH_TAG
```
Comment (robot9706) [inline web/crux-ui/e2e/with-login/nodes-deploy.spec.ts:15]: In this case there's no need for `IMAGE`, it can be replaced with `NGINX_TEST_IMAGE_WITH_TAG`.

**Review by robot9706 — APPROVED** [review]
Looks cool ✨, wait for another review and e2e.

**Comment by AmorfEvo** [comment]
LGTM (tests run with 4 workers)

**Review by robot9706 — APPROVED** [review]

**Review by AmorfEvo — APPROVED** [review]

**Pull request merged**